### PR TITLE
Split keyIDMasternode into keyIDOwner/keyIDOperator/keyIDVoting

### PR DESCRIPTION
**Pull request by codablock**
This is extracted from #2083. It can stand on its own and without all the other DIP3 code, it will simply default to using the same key for all 3 new keys.

### CODE REVIEW

**Comment by codablock** [comment]
Also added https://github.com/dashpay/dash/pull/2248/commits/acf820312b76209f7be9ba7fd9ec2cda767324fc

File: src/rpc/governance.cpp [hunk 1]
```diff
@@ -236,7 +236,7 @@ UniValue gobject_submit(const JSONRPCRequest& request)
 
     bool fMnFound = mnodeman.Has(activeMasternodeInfo.outpoint);
 
-    DBG( std::cout << "gobject: submit activeMasternodeInfo.keyIDMasternode = " << activeMasternodeInfo.keyIDMasternode.ToString()
+        DBG( std::cout << "gobject: submit activeMasternodeInfo.keyIDOperator = " << activeMasternodeInfo.keyIDOperator.ToString()
```
Comment (UdjinM6) [inline src/rpc/governance.cpp]: nit: whitespaces

File: src/rpc/governance.cpp [hunk 2]
```diff
@@ -286,7 +286,7 @@ UniValue gobject_submit(const JSONRPCRequest& request)
     if(govobj.GetObjectType() == GOVERNANCE_OBJECT_TRIGGER) {
         if(fMnFound) {
             govobj.SetMasternodeOutpoint(activeMasternodeInfo.outpoint);
-            govobj.Sign(activeMasternodeInfo.keyMasternode, activeMasternodeInfo.keyIDMasternode);
+                govobj.Sign(activeMasternodeInfo.keyOperator, activeMasternodeInfo.keyIDOperator);
```
Comment (UdjinM6) [inline src/rpc/governance.cpp]: same

File: src/rpc/governance.cpp [hunk 3]
```diff
@@ -367,6 +367,16 @@ UniValue gobject_vote_conf(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid vote outcome. Please use one of the following: 'yes', 'no' or 'abstain'");
     }
 
+    int govObjType;
```
Comment (UdjinM6) [inline src/rpc/governance.cpp]: this looks unrelated/unused

File: src/rpc/governance.cpp [hunk 4]
```diff
@@ -422,18 +432,28 @@ UniValue VoteWithMasternodeList(const std::vector<CMasternodeConfig::CMasternode
                                 const uint256& hash, vote_signal_enum_t eVoteSignal,
                                 vote_outcome_enum_t eVoteOutcome)
 {
+    int govObjType;
```
Comment (UdjinM6) [inline src/rpc/governance.cpp]: same

File: src/rpc/masternode.cpp [hunk 5]
```diff
@@ -1127,7 +1133,7 @@ UniValue masternodebroadcast(const JSONRPCRequest& request)
                 resultObj.push_back(Pair("outpoint", mnb.outpoint.ToStringShort()));
                 resultObj.push_back(Pair("addr", mnb.addr.ToString()));
                 resultObj.push_back(Pair("keyIDCollateralAddress", CBitcoinAddress(mnb.keyIDCollateralAddress).ToString()));
-                resultObj.push_back(Pair("keyIDMasternode", CBitcoinAddress(mnb.keyIDMasternode).ToString()));
+                resultObj.push_back(Pair("keyIDMasternode", CBitcoinAddress(mnb.keyIDOperator).ToString()));
```
Comment (UdjinM6) [inline src/rpc/masternode.cpp]: `keyIDMasternode`? Is this intentional (for backwards compatibility) or ..?

File: src/evo/providertx.cpp [hunk 6]
```diff
@@ -76,6 +76,16 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
     if (!ptx.scriptPayout.IsPayToPublicKeyHash())
         return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
 
+    CTxDestination payoutDest;
+    if (!ExtractDestination(ptx.scriptPayout, payoutDest)) {
+        // should not happen as we checked script types before
+        return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
```
Comment (UdjinM6) [inline src/evo/providertx.cpp]: Should probably alter rejection reason a bit so that it would be easier to debug i.e. `bad-protx-payee-dest` or smth like that

File: src/evo/providertx.cpp [hunk 7]
```diff
@@ -76,6 +76,16 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
...
+    }
+    // don't allow reuse of keys for different purposes
+    if (payoutDest == CTxDestination(ptx.keyIDOwner) || payoutDest == CTxDestination(ptx.keyIDOperator) || payoutDest == CTxDestination(ptx.keyIDVoting)) {
+        return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
```
Comment (UdjinM6) [inline src/evo/providertx.cpp]: same, e.g. `bad-protx-payee-reuse`

File: src/evo/providertx.cpp [hunk 8]
```diff
@@ -76,6 +76,16 @@ bool CheckProRegTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CValid
     if (!ptx.scriptPayout.IsPayToPublicKeyHash())
         return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
 
+    CTxDestination payoutDest;
+    if (!ExtractDestination(ptx.scriptPayout, payoutDest)) {
+        // should not happen as we checked script types before
+        return state.DoS(10, false, REJECT_INVALID, "bad-protx-payee");
+    }
+    // don't allow reuse of keys for different purposes
```
Comment (UdjinM6) [inline src/evo/providertx.cpp]: The comment here is a bit confusing imo. This doesn't allow reuse of keys which control payout address for other purposes, not reuse of keys in general.

File: src/rpc/masternode.cpp (hunk 5 above)
Comment (codablock) [reply src/rpc/masternode.cpp]: Yepp. This RPC will become unused after spork15 activation and at the same time I assume external tools are using this

**Comment by codablock** [comment]
@UdjinM6 Forgot to bump CMasternodeMan version string, pushed this now
